crates/astria-composer/src/searcher/executor.rs

```diff
@@ -176,18 +227,75 @@ impl Executor {
             .await
             .wrap_err("failed retrieving initial nonce from sequencer")?;
 
-        while let Some(bundle) = self.executor_rx.recv().await {
-            if let Err(e) = self.sign_and_submit(bundle).await {
-                // FIXME: currently this will fail both when there is an issue with the nonce and
-                // when unable to reach the sequencer. As there is currently no error returned by
-                // the sequencer for invalid nonces, there is nothing to handle. This should be
-                // changed after #364 is merged in a followup PR to handle nonce failues.
-                error!(
-                    error.message = %e,
-                    error.cause_chain = ?e,
-                    "failed submitting the bundle to the sequencer; bailing",
-                );
-                break;
+        let mut submission_fut = Fuse::terminated();
+        let mut nonce_fut = Fuse::terminated();
+        loop {
+            select! {
+                // receive new bundle for processing
+                Some(bundle) = self.executor_rx.recv(), if nonce_fut.is_terminated() && submission_fut.is_terminated() => {
+                    debug!(bundle = ?bundle, "executor received bundle for processing");
```

Comment:
Please don't do debug logging, even in `debug!` events. They break the json subscriber.